WordPress/src/main/java/org/wordpress/android/ui/domains/management/newdomainsearch/composable/NewDomainSearchScreen.kt

```diff
@@ -163,39 +166,50 @@ fun TransferDomainFooter(
 @Composable
 fun Domain(
     domain: ProposedDomain,
+    onDomainTapped: (domain: ProposedDomain) -> Unit,
     modifier: Modifier = Modifier
 ) {
-    Column(modifier = modifier.padding(16.dp)) {
-        Row {
-            Text(
-                text = domain.domainPrefix,
-                style = MaterialTheme.typography.bodyLarge
-            )
-            Text(
-                text = domain.domainSuffix,
-                style = MaterialTheme.typography.bodyLarge,
-                fontWeight = FontWeight.Bold
-            )
-        }
-        if (domain.salePrice.isNullOrEmpty()) {
-            Text(text = stringResource(id = R.string.new_domain_search_screen_list_item_regular_price, domain.price))
-        } else {
+    Surface(onClick = { onDomainTapped(domain) }) {
```

Comment:
In that case `modifier` should be used in the Surface instead. In addition, I'd suggest to add `modifier.fillMaxWidth()` to make the entire row clickable.

```suggestion
    Surface(
        onClick = { onDomainTapped(domain) },
        modifier = modifier.fillMaxWidth()
    ) {
```